programs/marinade-finance/src/instructions/user/withdraw_stake_account.rs

```diff
@@ -0,0 +1,302 @@
+        );
+
+        // burn mSOL
+        burn(
+            CpiContext::new(
+                self.token_program.to_account_info(),
+                Burn {
+                    mint: self.msol_mint.to_account_info(),
+                    from: self.burn_msol_from.to_account_info(),
+                    authority: self.burn_msol_authority.to_account_info(),
+                },
+            ),
+            msol_amount,
+        )?;
+        self.state.on_msol_burn(msol_amount)?;
+
+        // split deactivate split_lamports from stake account
+        msg!(
+            "Deactivate split {} ({} lamports) from stake {}",
+            self.split_stake_account.key(),
+            split_lamports,
+            stake.stake_account
+        );
+
+        let split_instruction = stake::instruction::split(
+            self.stake_account.to_account_info().key,
+            self.stake_withdraw_authority.key,
+            split_lamports,
+            &self.split_stake_account.key(),
+        )
+        .last()
+        .unwrap()
+        .clone();
+        invoke_signed(
+            &split_instruction,
+            &[
+                self.stake_program.to_account_info(),
+                self.stake_account.to_account_info(),
+                self.split_stake_account.to_account_info(),
+                self.stake_withdraw_authority.to_account_info(),
+            ],
+            &[&[
+                &self.state.key().to_bytes(),
+                StakeSystem::STAKE_DEPOSIT_SEED,
+                &[self.state.stake_system.stake_deposit_bump_seed],
+            ]],
+        )?;
+
+        solana_deactivate_stake(CpiContext::new_with_signer(
+            self.stake_program.to_account_info(),
+            SolanaDeactivateStake {
+                stake: self.split_stake_account.to_account_info(),
+                staker: self.stake_withdraw_authority.to_account_info(),
+                clock: self.clock.to_account_info(),
+            },
+            &[&[
+                &self.state.key().to_bytes(),
+                StakeSystem::STAKE_DEPOSIT_SEED,
+                &[self.state.stake_system.stake_deposit_bump_seed],
+            ]],
+        ))?;
+
+        stake.last_update_delegated_lamports -= split_lamports;
+
+        // we now consider amount no longer "active" for this specific validator
+        validator.active_balance -= split_lamports;
+        // and in state totals,
+        self.state.validator_system.total_active_balance -= split_lamports;
+
+        // update stake-list & validator-list
+        self.state.stake_system.set(
+            &mut self.stake_list.data.as_ref().borrow_mut(),
+            stake_index,
+            stake,
+        )?;
+        self.state.validator_system.set(
+            &mut self.validator_list.data.as_ref().borrow_mut(),
+            validator_index,
+            validator,
+        )?;
+
+        // assign user as withdrawer (owner) & staker for the split_stake_account
+        invoke(
```

Comment:
invoke_signed because you need to sign by PDA

Reply:
done.